Content.Client/Audio/AmbientSoundSystem.cs

```diff
@@ -0,0 +1,178 @@
+using System.Collections.Generic;
+using System.Linq;
+using Content.Shared.Audio;
+using Content.Shared.CCVar;
+using Robust.Client.Player;
+using Robust.Shared.Audio;
+using Robust.Shared.Configuration;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Map;
+using Robust.Shared.Player;
+using Robust.Shared.Random;
+
+namespace Content.Client.Audio
+{
+    /// <summary>
+    /// Samples nearby <see cref="AmbientSoundComponent"/> and plays audio.
+    /// </summary>
+    public sealed class AmbientSoundSystem : SharedAmbientSoundSystem
+    {
+        [Dependency] private IEntityLookup _lookup = default!;
+        [Dependency] private readonly IPlayerManager _playerManager = default!;
+        [Dependency] private readonly IRobustRandom _random = default!;
+
+        private int _maxAmbientCount;
+
+        private float _maxAmbientRange;
+        private float _cooldown;
+        private float _accumulator;
+
+        /// <summary>
+        /// How many times we can be playing 1 particular sound at once.
+        /// </summary>
+        private int _maxSingleSound = 3;
+
+        private Dictionary<AmbientSoundComponent, (IPlayingAudioStream? Stream, string Sound)> _playingSounds = new();
+
+        private const float RangeBuffer = 0.1f;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            var configManager = IoCManager.Resolve<IConfigurationManager>();
+            configManager.OnValueChanged(CCVars.AmbientCooldown, SetCooldown, true);
+            configManager.OnValueChanged(CCVars.MaxAmbientSources, SetAmbientCount, true);
+            configManager.OnValueChanged(CCVars.AmbientRange, SetAmbientRange, true);
+        }
+
+        private void SetCooldown(float value) => _cooldown = value;
+        private void SetAmbientCount(int value) => _maxAmbientCount = value;
+        private void SetAmbientRange(float value) => _maxAmbientRange = value;
+
+        public override void Shutdown()
+        {
+            base.Shutdown();
+            var configManager = IoCManager.Resolve<IConfigurationManager>();
+            configManager.UnsubValueChanged(CCVars.AmbientCooldown, SetCooldown);
+            configManager.UnsubValueChanged(CCVars.MaxAmbientSources, SetAmbientCount);
+            configManager.UnsubValueChanged(CCVars.AmbientRange, SetAmbientRange);
+        }
+
+        private int PlayingCount(string countSound)
```

Comment:
how exactly does this work with SoundCollections, since SoundSpecifier can be either a path or collection? this is just a string, so

Reply:
It was more about preventing a specific sound to avoid it being too repetitive; if it's in the same collection should eb okay IMO.